h/api/groups/logic.py

```diff
@@ -0,0 +1,30 @@
+from h.api import models
+
+
+def set_group_if_reply(annotation):
```

Comment:
Hmm. I prefer to enforce this server-side. If I remove it then people could use the API (or other clients etc) to create replies in different groups from the parents, even if our own UI doesn't let you do that.

I'd ask the question the other way round: do you see a reason why it _should_ be possible for a reply to belong to a different group? I don't see any benefit to us enabling that, it's not in any of our user stories.

The more consistent we can make our data, and the more we can enforce the consistency server-side, the easier it's gonna be to write code to handle that data, both our own server-side code and any API client code by us or third parties. There's just less variation and different possibilities to have to deal with. Surprising configurations like a reply that has a different group from its parent are especially beneficial to avoid for this reason.

Not having to worry about replies whose parent is in a different group and how to handle those (when the view is currently filtered to one group or the other, when the user has permission to read one group but not the other...) makes code easier to write.

I know we have a way of showing a reply without its parent already (the _Message not available_) but I'd like to minimise how possible it is for that to happen, users see it as a bug.